File: src/main/java/com/gusrylmubarok/reddit/backend/service/MailContentBuilder.java
package com.gusrylmubarok.reddit.backend.service;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

@Service
@AllArgsConstructor
public class MailContentBuilder {
    private final TemplateEngine templateEngine;
    // Will build HTML for send request activation account
    String build(String message) {
        Context context = new Context();
        context.setVariable("message", message);
        return templateEngine.process("mailTemplate", context);
    }

}

File: src/main/kotlin/com/reddot/service/mailer/MailContentBuilder.kt
package com.reddot.service.mailer

import org.springframework.stereotype.Service
import org.thymeleaf.TemplateEngine
import org.thymeleaf.context.Context

@Service
class MailContentBuilder (
    val templateEngine: TemplateEngine
) {

    fun build(message: String): String {
        val context: Context = Context()
        context.setVariable("message", message)

        return templateEngine.process("mailTemplate", context)
    }

}